File: string/1592_rearrange_spaces_between_words/1592_rearrange_spaces_between_words.go
package main

import (
	"fmt"
	"strings"
)

func reorderSpaces(text string) string {
	parts := strings.Split(text, " ")
	words := make([]string, 0)
	for _, part := range parts {
		if part != "" {
			words = append(words, part)
		}
	}
	if len(words) == 1 {
		return words[0] + strings.Repeat(" ", len(parts)-1)
	}
	join := strings.Repeat(" ", (len(parts)-1)/(len(words)-1))
	return strings.Join(words, join) + strings.Repeat(" ", (len(parts)-1)%(len(words)-1))
}

func main() {
	text := "  this   is  a sentence "
	res := reorderSpaces(text)
	fmt.Println(res)
}

File: string/1592_rearrange_spaces_between_words/1592_rearrange_spaces_between_words.py
class Solution:
    def reorderSpaces(self, text: str) -> str:
        cnt=text.count(" ")
        n = text.split()
        if len(n)==1:
            return text.strip()+cnt*" "
        space,last = divmod(cnt,len(n)-1)
        return (" "*space).join(n)+" "*last


text = "  this   is  a sentence "
res = Solution().reorderSpaces(text)
print(res)